src/lib/components/Header/Nav/NavItem.svelte

```diff
@@ -0,0 +1,108 @@
+  import chunk from "$lib/util/chunk";
+  import Link from "$lib/components/Link";
+
+  type $$Props = NavItemProps;
+
+  export let id = "";
+  export let link = "";
+  export let isCurrent = false;
+  export let megaMenuColumns = 0;
+  export let children: NavItemType[] = [];
+
+  let isParent = children.length > 0;
+  let isExpanded = false;
+
+  $: buttonClassNames = classNames(
+    "usa-accordion__button usa-nav__link",
+    isCurrent && "usa-current"
+  );
+  $: navLinkClassNames = classNames("usa-nav-link", isCurrent && "usa-current");
+
+  // Divide nav items into columns.
+  // TODO: We may want to let the CMS determine the position of each item in the future.
+  let megaMenu: NavItemType[][] = [];
+  if (megaMenuColumns > 0) {
+    const columnLength = Math.ceil(children.length / megaMenuColumns);
+    megaMenu = chunk(children, columnLength);
+  }
+
+  const toggleSubMenu = (show?: boolean) => {
+    if (typeof show !== "undefined") isExpanded = show;
+    else isExpanded = !isExpanded;
+  };
+
+  // By default focusout on the container will trigger for all its children, but using 'self'
+  //   doesn't resolve the issue since the container element isn't the one with the focus.
+  // This workaround will ignore the event if the new focused element is a child of the container.
+  // Based on this REPL: https://svelte.dev/repl/4c5dfd34cc634774bd242725f0fc2dab?version=3.46.4
+  const handleDropdownFocusLoss = ({
+    relatedTarget,
+    currentTarget,
+  }: FocusEvent & { currentTarget: EventTarget & HTMLDivElement }) => {
+    if (relatedTarget instanceof HTMLElement && currentTarget?.contains(relatedTarget)) return;
+    toggleSubMenu(false);
+  };
+
+  // We can't use on:click since it only triggers if the mousedown and mouseup events occur on the
+  //   same target, and on mobile the collapse of an accordion when it loses focus happens on
+  //   mousedown, sometimes shifting the elements so that mouseup misses the target.
+  // Instead, we'll use both mousedown and keydown to ensure everything functions accessibly on
+  //   both desktop and mobile screen sizes.
+  const handleDropdownKeyDown = (event: KeyboardEvent) => {
+    if (event.key === "Enter" || event.key === " ") toggleSubMenu();
+  };
+</script>
+
+{#if isParent}
+  <div on:focusout={handleDropdownFocusLoss}>
+    <button
+      type="button"
+      class={buttonClassNames}
+      aria-expanded={isExpanded}
+      aria-controls="extended-mega-nav-section-{id}"
+      on:keydown={handleDropdownKeyDown}
+      on:mousedown={() => toggleSubMenu()}
+    >
+      <!-- Extra <span/> elements in the this template are necessary for expand icons to load. -->
+      <span><slot /></span>
+    </button>
+    {#if megaMenu.length > 0}
+      <!-- Mega Menu Layout-->
+      <div
+        id="extended-mega-nav-section-{id}"
+        class="usa-nav__submenu usa-megamenu"
+        hidden={!isExpanded}
+      >
+        <div class="grid-row grid-gap-4">
```

Comment:
Wasn't sure if the `gap` here should be calculated based on the number of columns, so I just left this as the same in the example.

Reply:
This is still relevant after the latest updates, but it's been moved: https://github.com/la-ldaf/ldaf-site/blob/8204289d1dc87b38f6895769a6a8f53e09382b55/src/lib/components/Header/Nav/NavMenu.svelte#L78